packages/squiggle-lang/src/rescript/Utility/E/E_A.res

```diff
@@ -305,56 +305,50 @@ module Floats = {
     /*
       This function goes through a sorted array and divides it into two different clusters:
       continuous samples and discrete samples. The discrete samples are stored in a mutable map.
-      Samples are thought to be discrete if they have any duplicates.
+      Samples are thought to be discrete if they have at least `minDiscreteWight` duplicates.
+
+      If the min discreet weight is 4, that would mean that at least four elements needed from a specific
+      value for that to be kept as discrete. This is important because in some cases, we can expect that
+      some common elements will be generated by regular operations. The final continous array will be sorted.
```

Comment:
continuous* (Doesn't matter much here, but noticed)

Reply:
I might also flag in the comments that this code is optimized for performance instead of elegance. Maybe warn people not to try to optimize for elegance.

Reply:
I added a comment about the this being performance-critical, but tbh I don't really see the inelegance... I probably wrote too much imperative code like this in my life (but I guess I can see how it's icky if you're aiming for pure functional style).

The bit that annoys me the most about this is `FloatFloatMap` when it should be `FloatIntMap`, but that can't be fixed without larger changes.